### PR TITLE
Move deploy script to run

### PR DESCRIPTION
**Pull request by gmrabian**
### Description

Migrate our cloud deploy script to the run tool to align with the run command pattern
- Update github workflow to call deploy through the run command
- Remove custom deploy bash script
- Add deploy script functionality to the run.ts tool

There should be no change to the deploy commands and process, though the deploy logs will now be grouped under headings like "Installing Dependencies" and "SLS Deploy"

### Related ticket(s)

CMDCT-3645

---
### How to test

- Verify this branch deployed successfully

This is not for local dev! You can continue to use `./run local` for that

---
### Pre-review checklist

- ~[ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary~
- ~[ ] I have updated relevant documentation, if necessary~
- [x] I have performed a self-review of my code
- [x] I have manually tested this PR in the deployed cloud environment

### CODE REVIEW

File: src/run.ts [hunk 1]
```diff
@@ -137,6 +147,23 @@ async function run_all_locally() {
...
+      ["yarn", "install", "--frozen-lockfile"],
+      `services/${service}`
+    );
+  }
+}
+
+async function deploy(options: { stage: string }) {
+  const runner = new LabeledProcessRunner();
+  await install_deps_for_services(runner);
+  var deployCmd = ["sls", "deploy", "--stage", options.stage];
```
Comment (bangbay-bluetiger) [inline src/run.ts:163]: `const` instead of `var`.

File: src/run.ts [hunk 2]
```diff
@@ -137,6 +147,23 @@ async function run_all_locally() {
   run_fe_locally(runner);
 }
 
+async function install_deps_for_services(runner: LabeledProcessRunner) {
+  for (const service of deployedServices) {
+    await runner.run_command_and_output(
+      `Installing Dependencies`,
```
Comment (bangbay-bluetiger) [inline src/run.ts:153]: Does `Installing Dependencies` need to be in backticks?

File: src/run.ts [hunk 3]
```diff
@@ -137,6 +147,23 @@ async function run_all_locally() {
...
+    );
+  }
+}
+
+async function deploy(options: { stage: string }) {
+  const runner = new LabeledProcessRunner();
+  await install_deps_for_services(runner);
+  var deployCmd = ["sls", "deploy", "--stage", options.stage];
+  await runner.run_command_and_output(`SLS Deploy`, deployCmd, ".");
+}
```
[the remark this answers is not in the document]
Comment (gmrabian) [reply src/run.ts:165]: it does not! will change

File: src/run.ts (hunk 2 above)
Comment (gmrabian) [reply src/run.ts]: it does not! will change

**Review by bangbay-bluetiger — APPROVED** [review]

**Pull request merged**